app/controllers/devise/registrations_controller.rb

```diff
@@ -83,9 +83,10 @@ def cancel
   protected
 
   def update_needs_confirmation?(resource, previous)
-    resource.respond_to?(:pending_reconfirmation?) &&
-      resource.pending_reconfirmation? &&
-      previous != resource.unconfirmed_email
+    resource.respond_to?(:can_send_reconfirmation?) &&
```

Comment:
@SaiVardhan our new toggle shouldn't affect this logic, the reconfirmation works even when using the old email template. Let's go back to use `pending_reconfirmation?` instead of `can_send_reconfirmation?` here.